gnomad/assessment/summary_stats.py

```diff
@@ -452,3 +451,168 @@ def default_generate_gene_lof_matrix(
+            pop_expr,
+            hl.agg.count_where(
+                (csq_filter_expr) & (num_homs_expr == 0) & (num_hets_expr > 0)
+            ),
+        ),
+        f"pop_obs_hom_{csq_name}": hl.agg.group_by(
+            pop_expr, hl.agg.count_where((csq_filter_expr) & (num_homs_expr > 0)),
+        ),
+        f"pop_defined_{csq_name}": hl.agg.group_by(
+            pop_expr, hl.agg.count_where((csq_filter_expr) & (defined_sites_expr > 0)),
+        ),
+    }
+
+
+def default_generate_gene_lof_summary(
+    mt: hl.MatrixTable,
+    collapse_indels: bool = False,
+    tx: bool = False,
+    lof_csq_set: Set[str] = LOF_CSQ_SET,
+    meta_root: str = "meta",
+    pop_field: str = "pop",
+) -> hl.Table:
+    """
+    Generates summary counts for loss-of-function (LoF), missense, and synonymous variants.
+
+    Also calculates p, proportion of of haplotypes carrying a probable LoF (pLoF) variant, 
+    and observed/expected (OE) ratio of samples with homozygous pLoF variant calls.
+
+    Summary counts are (all per gene):
+        - Number of samples with no pLoF variants.
+        - Number of samples with heterozygous pLoF variants.
+        - Number of samples with homozygous pLoF variants.
+        - Total number of sites with genotype calls.
+        - All of the above stats grouped by population.
+
+    Assumes MT was created using `default_generate_gene_lof_matrix`.
+
+    :param mt: Input MatrixTable.
+    :param collapse_indels: Whether to collapse indels. Default is False.
+    :param tx: Whether input MT has transcript expression data. Default is False.
+    :param lof_csq_set: Set containing LoF transcript consequence strings. Default is LOF_CSQ_SET.
+    :param meta_root: String indicating top level name for sample metadata. Default is 'meta'.
+    :param pop_field: String indiciating field with sample population assignment information. Default is 'pop'.
+    :return: Table with het/hom summary counts.
+    """
+    if collapse_indels:
+        grouping = ["gene_id", "gene", "most_severe_consequence"]
+        if tx:
+            grouping.append("expressed")
+        else:
+            grouping.extend(["transcript_id", "canonical"])
+        mt = (
+            mt.group_rows_by(*grouping)
+            .aggregate_rows(
+                n_sites=hl.agg.sum(mt.n_sites),
+                n_sites_array=hl.agg.array_sum(mt.n_sites_array),
+                classic_caf=hl.agg.sum(mt.classic_caf),
+                max_af=hl.agg.max(mt.max_af),
+                classic_caf_array=hl.agg.array_sum(mt.classic_caf_array),
+            )
+            .aggregate_entries(
+                num_homs=hl.agg.sum(mt.num_homs),
+                num_hets=hl.agg.sum(mt.num_hets),
+                defined_sites=hl.agg.sum(mt.defined_sites),
+            )
+            .result()
+        )
+
+    ht = mt.annotate_rows(
+        **get_het_hom_summary_dict(
+            csq_name="lof",
+            csq_set=lof_csq_set,
```

Comment:
Is there a loftee filter somewhere? I don't see it in this code, but maybe it was filtered upstream. Apologies if I missed it

Reply:
`default_generate_gene_lof_matrix` upstream filters to loftee HC and no flags by default. I can also add the option here if you think it makes sense?

Reply:
Do you run these one after the other? If so, yes. But might be worth the option too. Not sure